From 251b4e4828c744e54d8952766bc8f3b32ce065a9 Mon Sep 17 00:00:00 2001
From: Bert Belder <bertbelder@gmail.com>
Date: Sun, 18 Dec 2022 12:50:43 +0100
Subject: [PATCH 4/4] Emit warning instead of error when builtins PGO profiles
 are out of date

This works around a build failure seen in V8 11.0.226.2 on some
platforms. Note that PGO data was just added by the V8 team in the last
commit before the release, so it doesn't seem likely that it is actually
out of date.

```
[1952/1960] ACTION //v8:run_mksnapshot_default(//build/toolchain/linux:clang_x64)
FAILED: gen/v8/embedded.S gen/v8/snapshot.cc
python ../../../../v8/tools/run.py ./mksnapshot --turbo_instruction_scheduling --target_os=linux --target_arch=x64 --embedded_src gen/v8/embedded.S --turbo-profiling-input ../../../../v8/tools/builtins-pgo/x64.profile --abort-on-bad-builtin-profile-data --embedded_variant Default --random-seed 314159265 --startup_src gen/v8/snapshot.cc --no-native-code-counters
#
# Fatal error in , line 0
# Rejected profile data for StringEqual due to function change. Please use tools/builtins-pgo/generate.py to refresh it.
#
#
#
#FailureMessage Object: 0x7ffed697ff60
==== C stack trace ===============================

    ./mksnapshot(+0x162e8e3) [0x561c9db3e8e3]
    ./mksnapshot(+0x162e11b) [0x561c9db3e11b]
    ./mksnapshot(+0x1623185) [0x561c9db33185]
    ./mksnapshot(+0xda6550) [0x561c9d2b6550]
    ./mksnapshot(+0xc42f7e) [0x561c9d152f7e]
    ./mksnapshot(+0xf438f3) [0x561c9d4538f3]
    ./mksnapshot(+0xf2a2c0) [0x561c9d43a2c0]
    ./mksnapshot(+0x4e80bc) [0x561c9c9f80bc]
    ./mksnapshot(+0x4e6e22) [0x561c9c9f6e22]
    ./mksnapshot(+0x3c9641) [0x561c9c8d9641]
    ./mksnapshot(+0x8dc554) [0x561c9cdec554]
    ./mksnapshot(+0x3c4357) [0x561c9c8d4357]
    /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf3) [0x7f36e1645083]
    ./mksnapshot(_start+0x2e) [0x561c9c8d111e]
Return code is -5
ninja: build stopped: subcommand failed.
```
---
 BUILD.gn | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/BUILD.gn b/BUILD.gn
index 19399e86e8..a67bac95c1 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -2201,7 +2201,7 @@ template("run_mksnapshot") {
 
       # Replace this with --warn-about-builtin-profile-data to see the full
       # list of builtins with incompatible profiles.
-      args += [ "--abort-on-bad-builtin-profile-data" ]
+      args += [ "--warn-about-builtin-profile-data" ]
     }
 
     # This is needed to distinguish between generating code for the simulator
-- 
2.37.3

